Neos.ContentRepository.Core/Tests/Unit/NodeType/NodeTypeManagerTest.php

```diff
@@ -335,4 +334,92 @@ public function getSubNodeTypesWithDifferentIncludeFlagValuesReturnsCorrectValue
         self::assertArrayNotHasKey('Neos.ContentRepository.Testing:AbstractType', $subNodeTypes);
     }
+
+    /**
+     * @test
+     */
+    public function anInheritedNodeTypePropertyCannotBeUnset(): void
+    {
+        $nodeTypesFixture = [
+            'Neos.ContentRepository.Testing:Base' => [
+                'properties' => [
+                    'foo' => [
+                        'type' => 'boolean',
+                    ]
+                ]
+            ],
+            'Neos.ContentRepository.Testing:Sub' => [
+                'superTypes' => ['Neos.ContentRepository.Testing:Base' => true],
+                'properties' => [
+                    'foo' => null
+                ]
+            ]
+        ];
+
+        $this->prepareNodeTypeManager($nodeTypesFixture);
+        $nodeType = $this->nodeTypeManager->getNodeType('Neos.ContentRepository.Testing:Sub');
+
+        self::assertSame(['foo' => ['type' => 'boolean']], $nodeType->getProperties());
+    }
+
+    /**
+     * @test
+     */
+    public function allInheritedNodeTypePropertiesCannotBeUnset(): void
+    {
+        $nodeTypesFixture = [
+            'Neos.ContentRepository.Testing:Base' => [
+                'properties' => [
+                    'foo' => [
+                        'type' => 'boolean',
+                    ]
+                ]
+            ],
+            'Neos.ContentRepository.Testing:Sub' => [
+                'superTypes' => ['Neos.ContentRepository.Testing:Base' => true],
+                'properties' => null
+            ]
+        ];
+
+        $this->prepareNodeTypeManager($nodeTypesFixture);
+        $nodeType = $this->nodeTypeManager->getNodeType('Neos.ContentRepository.Testing:Sub');
+
+        self::assertSame(['foo' => ['type' => 'boolean']], $nodeType->getProperties());
+    }
+
+    /**
+     * @test
+     */
+    public function anInheritedNodeTypePropertyCannotBeSetToEmptyArray(): void
+    {
+        $nodeTypesFixture = [
+            'Neos.ContentRepository.Testing:Base' => [
+                'properties' => [
+                    'foo' => [
+                        'type' => 'boolean',
+                        'ui' => [
+                            'inspector' => [
+                                'group' => 'things'
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+            'Neos.ContentRepository.Testing:Sub' => [
+                'superTypes' => ['Neos.ContentRepository.Testing:Base' => true],
+                'properties' => [
+                    // Pseudo unset.
+                    // The property will still be existent but looses its type information (falls back to string).
+                    // Also, the property will not show up anymore in the ui as the inspector configuration is gone as well.
+                    'foo' => []
+                ]
+            ]
```

Comment:
we should discuss at some point if we like this kinda stuff

Reply:
@mhsdesign do we have an issue where this can be discussed?